### PR TITLE
Add gradle script for publishing as Maven publication

### PR DESCRIPTION
**Pull request by aguibert**
Adds the `hibernate-rx-api` and `hibernate-rx-core` modules as publishable modules. This makes it so when you do `./gradlew publish` or `./gradlew publishToMavenLocal` the build will publish the maven artifacts to your local Maven repo.

There is now a common publish script at publish.gradle and any sub-module that wants to publish a maven artifact can do `apply from: publishScript` in their build.gradle file.

### CODE REVIEW

File: publish.gradle [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    if (performRelease) {
+        repositories {
+            maven {
+                url "https://oss.sonatype.org/service/local/staging/deploy/maven2"
+                credentials {
+                    username ossrhUsername
+                    password ossrhPassword
+                }
+            }
+        }
```
Comment (DavideD) [inline publish.gradle:28]: I'm not sure this is going to work. We cannot publish our username and password, It should be read from the settings.xml. Also we publish the artifact on our server and then it will take care to sync with Maven central.

In ORM they've just worked on the release scripts, we should check with @dreab8 and use a similar approach.

File: hibernate-rx-api/src/main/java/org/hibernate/rx/RxSession.java [hunk 2]
```diff
@@ -84,7 +84,7 @@
 	/**
 	 * Asynchronously persist the given transient instance, first assigning
```
Comment (DavideD) [inline hibernate-rx-api/src/main/java/org/hibernate/rx/RxSession.java:85]: Unless I'm missing something. Changes to the javadoc should be in a separate commit because they have nothing to do with the build.

File: publish.gradle (hunk 1 above)
Comment (DavideD) [reply publish.gradle:28]: @dreab8  could you link us to the part in the build of ORM that releases, please?

Comment (DavideD) [reply publish.gradle:28]: By the way, we an skeep the configuration of the remote repositories for now and only keep the local one.
We will add them later as a separate issues

Comment (dreab8) [reply publish.gradle:28]: @DavideD check  https://github.com/hibernate/hibernate-orm/blob/master/release/release.gradle and https://github.com/hibernate/hibernate-orm/blob/master/gradle/publishing-repos.gradle

Comment (dreab8) [reply publish.gradle:28]: you may be also interested in https://github.com/hibernate/hibernate-orm/blob/master/gradle/publishing-pom.gradle and https://github.com/hibernate/hibernate-orm/blob/master/gradle/published-java-module.gradle

Comment (DavideD) [reply publish.gradle:28]: THanks

File: hibernate-rx-api/src/main/java/org/hibernate/rx/RxSession.java (hunk 2 above)
Comment (aguibert) [reply hibernate-rx-api/src/main/java/org/hibernate/rx/RxSession.java:85]: I can move them into a separate commit. They were related to publishing the javadoc artifacts because I had to make the javadoc HTML5 compatible

**Comment by aguibert** [comment]
thanks for reviewing @DavideD, PR is updated with your review comments addressed

File: hibernate-rx-api/src/main/java/org/hibernate/rx/RxSession.java (hunk 2 above)
Comment (DavideD) [reply hibernate-rx-api/src/main/java/org/hibernate/rx/RxSession.java:85]: Thanks, it's still nice to have them in a separate commit so that we can explain why that change is required.

**Comment by DavideD** [comment]
Merged https://github.com/hibernate/hibernate-rx/pull/97

Thanks a lot!
